### PR TITLE
[INLONG-3996][Sort] Sort support all migrate for database

### LINKED ISSUE #3996: [Feature] Sort support all migrate for database
### Description

[Feature] Sort support all migrate for database

### Use case

[Feature] Sort support all migrate for database

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by EMsnap**
### Title Name: [INLONG-3996][Sort] Sort support all migrate for database

Fixes #3996 

### Motivation

Sort support all migrate for database

### Modifications

Sort support all migrate for database

### Documentation

  - Does this pull request introduce a new feature? (yes)

### CODE REVIEW

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/cdc/mysql/table/MySqlReadableMetadata.java [hunk 1]
```diff
@@ -96,26 +96,60 @@ public Object read(SourceRecord record) {
...
+                @org.jetbrains.annotations.Nullable TableChanges.TableChange tableSchema, RowData rowData) {
+
+                Struct messageStruct = (Struct) record.value();
+                Struct sourceStruct = messageStruct.getStruct(FieldName.SOURCE);
+                // tableName
+                String tableName = getMetaData(record, AbstractSourceInfo.TABLE_NAME_KEY);
+                // databaseName
+                String databaseName = getMetaData(record, AbstractSourceInfo.DATABASE_NAME_KEY);
+                // op_ts
+                long op_ts = (Long) sourceStruct.get(AbstractSourceInfo.TIMESTAMP_KEY);
```
Comment (healchow) [inline inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/cdc/mysql/table/MySqlReadableMetadata.java:125]: Maybe using the `opTs` is better?

File: inlong-sort/sort-formats/format-json/src/main/java/org/apache/inlong/sort/formats/json/canal/CanalJson.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    public String table;
+    public String type;
+    public String database;
+    public long ts;
+    public String sql;
+    public Map<String, String> mysqlType;
+    public boolean isDdl;
+    public List<String> pkNames;
+
+    public List<Map<String, Object>> getData() {
```
[the remark this answers is not in the document]
Comment (EMsnap) [reply inlong-sort/sort-formats/format-json/src/main/java/org/apache/inlong/sort/formats/json/canal/CanalJson.java:38]: lombok will create get method for boolean and jackson will then add a (ddl) property (should be isDdl so I did not use it

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/cdc/mysql/table/MySqlReadableMetadata.java (hunk 1 above)
Comment (EMsnap) [reply inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/cdc/mysql/table/MySqlReadableMetadata.java:125]: thx, make sense

**Review by GanfengTan — APPROVED** [review]

File: inlong-sort/sort-formats/format-json/src/main/java/org/apache/inlong/sort/formats/json/canal/CanalJson.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (healchow) [reply inlong-sort/sort-formats/format-json/src/main/java/org/apache/inlong/sort/formats/json/canal/CanalJson.java:38]: OK, I got it.

**Review by healchow — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]

**Pull request merged**